File: packages/js-dpp/lib/dataTrigger/DataTriggerExecutionContext.js
class DataTriggerExecutionContext {
  /**
   * @param {StateRepository} stateRepository
   * @param {Buffer|Identifier} ownerId
   * @param {DataContract} dataContract
   * @param {StateTransitionExecutionContext} stateTransitionExecutionContext
   */
  constructor(stateRepository, ownerId, dataContract, stateTransitionExecutionContext) {
    /**
     * @type {StateRepository}
     */
    this.stateRepository = stateRepository;
    this.ownerId = ownerId;
    this.dataContract = dataContract;
    this.stateTransitionExecutionContext = stateTransitionExecutionContext;
  }

  /**
   * @returns {StateRepository}
   */
  getStateRepository() {
    return this.stateRepository;
  }

  /**
   * @returns {Buffer|Identifier}
   */
  getOwnerId() {
    return this.ownerId;
  }

  /**
   * @returns {DataContract}
   */
  getDataContract() {
    return this.dataContract;
  }

  /**
   * @return {StateTransitionExecutionContext}
   */
  getStateTransitionExecutionContext() {
    return this.stateTransitionExecutionContext;
  }
}

module.exports = DataTriggerExecutionContext;

File: packages/rs-drive-abci/src/execution/data_trigger/data_trigger_execution_context.rs
use crate::platform::PlatformStateRef;
use dpp::{
    prelude::{DataContract, Identifier},
    state_repository::StateRepositoryLike,
    state_transition::state_transition_execution_context::StateTransitionExecutionContext,
};
use drive::drive::Drive;
use drive::grovedb::TransactionArg;

#[derive(Clone)]
pub struct DataTriggerExecutionContext<'a> {
    pub platform: &'a PlatformStateRef<'a>,
    pub transaction: TransactionArg<'a, 'a>,
    pub owner_id: &'a Identifier,
    pub data_contract: &'a DataContract,
    pub state_transition_execution_context: &'a StateTransitionExecutionContext,
}

impl<'a> DataTriggerExecutionContext<'a> {
    /// Creates a new instance of DataTriggerExecutionContext
    pub fn new(
        platform: &'a PlatformStateRef<'a>,
        transaction: TransactionArg,
        owner_id: &'a Identifier,
        data_contract: &'a DataContract,
        state_transition_execution_context: &'a StateTransitionExecutionContext,
    ) -> Self {
        DataTriggerExecutionContext {
            platform,
            transaction,
            owner_id,
            data_contract,
            state_transition_execution_context,
        }
    }
}
